src/fullscreen/fullscreen-controller.js

```diff
@@ -0,0 +1,255 @@
+//@flow
+import EventManager from '../event/event-manager';
+import Player from '../player';
+import FakeEvent from '../event/fake-event';
+import * as Utils from '../utils/util';
+
+/**
+ * The IOS fullscreen class name.
+ * @type {string}
+ * @const
+ */
+const IN_BROWSER_FULLSCREEN_FOR_IOS: string = 'playkit-in-browser-fullscreen-mode';
+
+/**
+ * @class FullscreenController
+ * @param {Player} player - The player.
+ */
+class FullscreenController {
+  _player: Player;
+  _inBrowserFullscreen: boolean;
+  _isInBrowserFullscreen: boolean;
+
+  /**
+   * after component mounted, set up event listeners to window fullscreen state change
+   * @param {Player} player - element to enter fullscreen
+   * @memberof FullScreenController
+   * @returns {void}
+   */
+  constructor(player: Player): void {
+    this._player = player;
+    this._inBrowserFullscreen = this._player.config.playback.inBrowserFullscreen;
+    //flag to cover the option that inBrowserFullscreen selected and we should know if it's full screen
+    this._isInBrowserFullscreen = false;
+    this.registerFullScreenEvents();
+  }
+
+  /**
+   * if native fullscreen mode
+   * @memberof FullScreenController
+   * @returns {boolean} - the current fullscreen state of the document
+   */
+  _isNativeFullscreen(): boolean {
+    const videoElement: ?HTMLVideoElement = typeof this._player.getVideoElement === 'function' ? this._player.getVideoElement() : null;
+    return (
+      (typeof document.fullscreenElement !== 'undefined' && Boolean(document.fullscreenElement)) ||
```

Comment:
```suggestion
      (Boolean(document.fullscreenElement)) ||
```
`Boolean(undefined) === false` No?
same for all lines below

Reply:
That what was before and we spoke about and agreed to keep it like that